api/db-cdr/generate-cdr/make-bq-data.sh

```diff
@@ -209,33 +209,80 @@ case when count_value > 0 then round(count_value/$person_count, 2)
+# domain info updates                    #
 ##########################################
 
-#Set the survey participant count
+# Set all_concept_count and standard_concept_count on domain_info
+bq --quiet --project=$BQ_PROJECT query --nouse_legacy_sql \
+"update \`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.domain_info\` d
+set d.all_concept_count = c.all_concept_count, d.standard_concept_count = c.standard_concept_count from
+(select c.domain_id as domain_id, COUNT(DISTINCT c.concept_id) as all_concept_count,
+SUM(CASE WHEN c.standard_concept IN ('S', 'C') THEN 1 ELSE 0 END) as standard_concept_count from
+\`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.concept\` c
+join \`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.domain_info\` d2
+on d2.domain_id = c.domain_id
+and (c.count_value > 0 or c.source_count_value > 0)
+group by c.domain_id) c
+where d.domain_id = c.domain_id
+"
+
+# Set participant counts for each domain
+bq --quiet --project=$BQ_PROJECT query --nouse_legacy_sql \
+"update \`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.domain_info\` d
+set d.participant_count = r.count_value from
+\`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.achilles_results\` r
+where r.analysis_id = 3000 and r.stratum_1 = CAST(d.concept_id AS STRING)
+and r.stratum_3 = d.domain_id
+and r.stratum_2 is null
+"
+
+##########################################
+# survey count updates                   #
+##########################################
+
+# Set the survey participant count on the concept
 bq --quiet --project=$BQ_PROJECT query --nouse_legacy_sql \
 "update \`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.concept\` c1
 set c1.count_value=count_val from
 (select count(distinct ob.person_id) as count_val,cr.concept_id_2 as survey_concept_id from \`${BQ_PROJECT}.${BQ_DATASET}.observation\` ob
 join \`${BQ_PROJECT}.${BQ_DATASET}.concept_relationship\` cr
-on ob.observation_source_concept_id=cr.concept_id_1 join \`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.db_domain\` dbd
-on cr.concept_id_2=dbd.concept_id
-where dbd.db_type='survey' and dbd.concept_id is not null
+on ob.observation_source_concept_id=cr.concept_id_1 join \`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.survey_module\` sm
+on cr.concept_id_2=sm.concept_id
 group by cr.concept_id_2)
 where c1.concept_id=survey_concept_id"
 
+# Set the participant count on the survey_module row
+bq --quiet --project=$BQ_PROJECT query --nouse_legacy_sql \
+"update \`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.survey_module\` sm
+set sm.participant_count=c.count_value from
+`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.concept\` c
+where c.concept_id=sm.concept_id"
+
+# Set the question count on the survey_module row
+bq --quiet --project=$BQ_PROJECT query --nouse_legacy_sql \
+"update \`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.survey_module\` sm
+set sm.question_count=num_questions from
+(select count(distinct cr.concept_id_1) num_questions, cr.concept_id_2 as survey_concept_id from
+`${BQ_PROJECT}.${BQ_DATASET}.concept_relationship\` cr
```

Comment:
Should this be Workbench_Project and dataset? This code would have to run after concept and concept_relationship is imported.  May not matter.

Reply:
I don't think it matters for this.